address/book.go

```diff
@@ -193,15 +194,70 @@ func NewBook(cfg BookConfig) *Book {
 	}
 }
 
+// queryAssetInfo attempts to locate asset genesis information by querying
+// geneses already known to this node. If asset issuance was not previously
+// verified, we then query universes in our federation for issuance proofs.
+func (b *Book) queryAssetInfo(ctx context.Context,
+	id asset.ID) (*asset.AssetGroup, error) {
+
+	// Check if we know of this asset ID already.
+	assetGroup, err := b.cfg.Store.QueryAssetGroup(ctx, id)
+	switch {
+	case assetGroup != nil:
+		return assetGroup, nil
+
+	// Asset lookup failed gracefully; continue to asset lookup using the
+	// AssetSyncer if enabled.
+	case errors.Is(err, ErrAssetGroupUnknown):
+		if !b.cfg.SyncerActive {
+			return nil, ErrAssetGroupUnknown
+		}
+
+	case err != nil:
+		return nil, err
+	}
+
+	log.Debugf("asset %v is unknown, attempting to bootstrap", id.String())
+
+	// Use the AssetSyncer to query our universe federation for the asset.
+	err = b.cfg.Syncer.FetchAssetInfo(ctx, &id)
+	if err != nil {
+		return nil, err
+	}
+
+	// The asset genesis info may have been synced from a universe
+	// server; query for the asset ID again.
+	assetGroup, err = b.cfg.Store.QueryAssetGroup(ctx, id)
+	if err != nil {
+		return nil, err
+	}
+
+	log.Debugf("bootstrap succeeded for asset %v", id.String())
+
+	// If the asset was found after sync, update our universe sync config
+	// to ensure that we sync future issuance proofs.
+	var groupKey *btcec.PublicKey
+	if assetGroup.GroupKey != nil {
+		groupKey = &assetGroup.GroupKey.GroupPubKey
+	}
+
+	err = b.cfg.Syncer.UpdateSyncConfig(ctx, &id, groupKey)
```

Comment:
I think we can get away with just a single method here that'll try to sync, and if it's able to, it'll add to the config. That way the addr book needs to know less information about syncing/configs, etc.

Reply:
Was thinking of that initially as well, but it's a bit awkward without adding group lookup to the various universe and federation universes (which I figured we didn't want to do).

From the context of `FetchAssetInfo`, when a sync succeeds, we don't know which proof is for the group anchor, and we would need to use that proof to return results equivalent to what `QueryAssetGroup` returns.

Reply:
Not sure I follow: why does it matter which proof gets returned here? IIUC, we just care that we have the genesis information for the asset ID in question. Assuming sync has completed, we should have the genesis info on disk, so we repeat the query that failed to get the info we need.

Reply:
For a group we also need the group key and the witness from the group anchor to match `QueryAssetGroup`. So those would only be in the group anchor proof.

Or do you mean perform the lookup again in `EnableAssetSync`?